### ISSUE TITLE
Flow not accepting Numpy array as parameter value

### ISSUE THREAD

**Issue by ntaylorwss** [PrefectHQ/prefect#832]
I thought I would begin using Prefect 0.5.0 by replicating a fairly simple (and mildly nonsensical) toy machine learning pipeline I had previously prepared using my tools:

![image](https://user-images.githubusercontent.com/14324416/54897944-aeb28f80-4e87-11e9-8148-14475e85bbb0.png)

That ended up looking like so:

```
@task
def RGBtoGrey(image, keep_dim=True):
    if len(X.shape) != 3:
        raise ValueError("Input image must have 3 dimensions")

    out = (X.max(axis=2) + X.min(axis=2)) / 2.
    
    if keep_dim:
        out = np.expand_dims(out, -1)
    return out

class OneHotRange(Task):
    def __init__(self, strict=False):
        self.strict = strict
        self.metadata = {}
        super().__init__()
    
    def fit(self, X):
        try:
            X = X.astype(int)
        except ValueError:
            raise TypeError("Array must be ints only")
        if not np.array_equal(X, X.astype(int)):
            raise TypeError("Array must be ints only")

        self.metadata['min_val'] = X.min()
        self.metadata['max_val'] = X.max()

    def transform(self, X):
        out = (np.arange(self.metadata['min_val'], self.metadata['max_val']+1) == X[..., None]) * 1
        if self.kwargs['strict'] and (out.sum(axis=-1) == 0).any():
            raise ValueError("New value encountered in transform that was not present in fit")
        return out
    
    def run(self, X):
        self.fit(X)
        return self.transform(X)
    

class KerasModel(Task):
    def __init__(self, model):
        self.model = model
        super().__init__()
        
    def fit(self, X, Y):
        self.model.fit(X, Y)
        
    def transform(self, X, Y):
        return self.model.predict(X)
    
    def run(self, X, Y):
        self.fit(X, Y)
        return self.transform(X)
    
    
with Flow("Keras") as flow:
    image1 = Parameter('image1')
    image2 = Parameter('image2')
    label = Parameter('label')
    bw_image1 = RGBtoGrey(image1)
    bw_image2 = RGBtoGrey(image2)
    ohe_label = OneHotRange()(label)
    model_pred = KerasModel(model)([bw_image1, bw_image2], ohe_label)
```

The Flow construction seemed to go fine. I ran into trouble when I attempted to run this Flow by passing the Numpy arrays in `data` as Parameter values:

```
>>> state = flow.run(parameters=data)
>>> state.result[image1]
Failed("Unexpected error: TypeError("Object of type 'ndarray' is not JSON serializable",)")
```

So it seems passing Numpy arrays directly is not going to work.

Is there a different approach I should be taking to this kind of Flow? Am I misusing Prefect in its current state by trying to tackle this kind of problem? What would be the next steps for a user in this situation?

**Comment by ntaylorwss** [comment]
One note I should make is that if I was to read these arrays from disk as an initial task, perhaps passing filenames as Parameters instead of Numpy arrays, everything goes off without a hitch. So this is one alternative that seems to work, but I'm curious if any other approaches would be recommended.

**Comment by jlowin** [comment]
@ntaylorwss, thanks as always for the complete issue report.  

The reason for the current behavior is that one of the primary uses of `Parameters` is to configure jobs via remote API, and so there's a JSON-able requirement for parameter values.

However, that's too constraining and we definitely want to support your use case. #733 will formalize a method for type-casting Parameter values. In the meantime, a quick work around is to turn "checkpointing" off, which will avoid `Parameter` value serialization. This isn't a `kwarg` for `Parameters` (it is for `Tasks` generally), so try the following:

```python
with Flow("Keras") as flow:
    image1 = Parameter('image1')
	image1.checkpoint = False
```

Now `flow.run(image1=<anything>)` should work.

Sorry for the inconvenience -- @cicdw we may want to consider a short-term point release that corrects some of these UX issues at launch.

**Comment by jlowin** [comment]
Also :100: for posting your viz.

**Comment by cicdw** [comment]
Closed via #839
